### PR TITLE
refactor: storage redesign

### PR DESCRIPTION
**Pull request by PaulRBerg**
Addresses https://github.com/cantinasec/review-sablier2/issues/13, the problem explained in https://github.com/PaulRBerg/prb-proxy/discussions/100, and [PRB-01M: Insufficient Protection Against Hostile Takeover](https://omniscia.io/reports/sablier-prb-proxy-647f15164f25e70014286487/manual-review/PRBProxy-PRB#PRB-01M).

Changelog:

- [x] The `plugins` and the `permissions` are now in the registry
- [x] There is no `Annex` contract anymore

### CODE REVIEW

File: src/PRBProxy.sol [hunk 1]
```diff
@@ -93,7 +89,8 @@ contract PRBProxy is
     /// @inheritdoc IPRBProxy
     function execute(address target, bytes calldata data) external payable override returns (bytes memory response) {
         // Check that the caller is either the owner or an envoy with permission.
-        if (owner != msg.sender && !permissions[msg.sender][target]) {
+        bool permission = registry.getPermissionByOwner({ owner: owner, envoy: msg.sender, target: target });
```
Comment (tibbarytsur) [inline src/PRBProxy.sol:92]: You can save gas here; check `msg.sender` is the owner before the call to the registry to check the permission of `msg.sender` as an envoy. It's not necessary to do this call for the owner as they should always have permission, unless you want to make it possible for an owner to block themselves from using certain target contracts.

**Review by andreivladbrg — APPROVED** [review]
Generally looks good

A small thing but I think this should be updated:
https://github.com/PaulRBerg/prb-proxy/blob/37dc8be6cb6eefce0b8199cdd8a942ee5f56350c/CHANGELOG.md?plain=1#L38

File: src/PRBProxy.sol (hunk 1 above)
Comment (PaulRBerg) [reply src/PRBProxy.sol:92]: Good catch!

We don't want to let the owner block themselves from using any targets, so we will make the call to `getPermissionByOwner` only after making the `owner` check.

**Comment by PaulRBerg** [comment]
Thanks for the review, both!

@andreivladbrg - the `CHANGELOG` should remain unchanged because its purpose is to document the historical evolution of the proxy.

**Pull request merged**